src/arena.rs

```diff
@@ -188,6 +188,10 @@ impl<T> Arena<T> {
 
     /// Adds a new value to the arena, returning a typed handle.
     pub fn append(&mut self, value: T) -> Handle<T> {
+        debug_assert!(self.data.len() <= u32::MAX as usize);
+        if self.data.len() as u32 == u32::MAX {
```

Comment:
let's instead change `new_unchecked` into just `new`, and have `expect` there. Less unsafe code!